spark/src/test/scala/ai/chronon/spark/test/JoinTest.scala

```diff
@@ -916,22 +651,24 @@ class JoinTest {
                                      | JOIN queries
                                      | ON queries.item <=> part.item AND queries.ts <=> part.ts AND queries.ds <=> part.ds
                                      |""".stripMargin)
-    expected.show()
-
+    if (logger.isDebugEnabled){
+      expected.show()
+    }
     val diff = Comparison.sideBySide(expected, computed, List("item", "ts", "ds"))
+    val diffCount = diff.count() 
     val queriesBare =
       tableUtils.sql(s"SELECT item, ts, ds from $itemQueriesTable where ds >= '$start' and ds <= '$dayAndMonthBefore'")
     assertEquals(queriesBare.count(), computed.count())
-    if (diff.count() > 0) {
-      println(s"Diff count: ${diff.count()}")
-      println(s"diff result rows")
+    if (diffCount > 0) {
+      logger.debug(s"Diff count: ${diffCount}")
+      logger.debug(s"diff result rows")
```

Comment:
let us use `warn` actually if the diff requires attention. it will be easier for debugging
```suggestion
      logger.warn(s"Diff count: ${diffCount}")
      logger.warn(s"diff result rows")
```